OrbitClientModel/CaptureSerializer.cpp

```diff
@@ -64,9 +66,15 @@ CaptureInfo GenerateCaptureInfo(
   for (const auto& address_info : capture_data.address_infos()) {
     orbit_client_protos::LinuxAddressInfo* added_address_info = capture_info.add_address_infos();
     added_address_info->CopyFrom(address_info.second);
-    // Fix names in address infos (some might only be in process):
-    added_address_info->set_function_name(
-        capture_data.GetFunctionNameByAddress(added_address_info->absolute_address()));
+    const orbit_client_protos::FunctionInfo* function =
+        capture_data.FindFunctionByAddress(added_address_info->absolute_address(), false);
```

Comment:
I'm curious is it possible to fix capture_data.address_infos before the serialization (e.g. when finalizing capture)? Then we don't need to fix names/offset/module during the serialization.

Reply:
I agree that we need a more conceptual fix. But I don't think we can easily patch the address info right now, due to synchronization. 
Address info might be queried also during a capture (due to selection of callstacks), so it can't be just fixed on capture synchronization. Also you might load symbols during a capture or after the capture. This all would make it not cleaner.

We need at some point a way to get rid of the "reads" from capture data during the capture (and rather stream data to the readers), as well as some clever way to handle the two sources for address information. 
For now, I would argue against doing a somewhat better, but more complicated solution, and go with this quick fix.